### ISSUE TITLE
Jaws Screen Reader Issues

### ISSUE THREAD

**Issue by mblanchette5** [microsoft/AzureStorageExplorer#7232]
### Preflight Checklist

- [X] I have installed the [latest version of Storage Explorer](https://github.com/Microsoft/AzureStorageExplorer/releases/latest).
- [X] I have checked existing resources, including the [troubleshooting guide](https://docs.microsoft.com/azure/storage/common/storage-explorer-troubleshooting) and the [release notes](https://github.com/microsoft/AzureStorageExplorer/releases).
- [X] I have searched for [similar issues](https://github.com/Microsoft/AzureStorageExplorer/issues).

### Storage Explorer Version

1.28.1

### Architecture

arm64

### Storage Explorer Build Number

20230309.1

### Platform

Windows

### Bug Description

I am trying to use Storage Explorer with a screen reader. 
 I am able to work thru the treeview and select a resource, but the screen reader is not speaking the contents to me.  When I do this in a Teams meeting, teammates are able to see files on the right side of the screen as expected.  I have tried both JAWS and Narrator with the same result.

### Steps to Reproduce

I am blind, and am trying to use Storage Explorer with a screen reader.  I am able to work thru the treeview on the left side of the screen, but after selecting a resource from the treeview, the screen reader is not speking the contents to me.  While running storage Explorer in a Teams meeting, my teammates are able to see files on the right side of my screen, but the screen reader isn't reading them to me.  The [Azure Storage Explorer Accessibility | Microsoft Learn](https://learn.microsoft.com/en-us/azure/vs-azure-tools-storage-explorer-accessibility) article mentions NVDA, which I  don't have.  I am using JAWS, and have also tried Narrator, with the same result.  Thank you.

### Actual Experience

I am trying to use Storage Explorer with a screen reader. 
 I am able to work thru the treeview and select a resource, but the screen reader is not speaking the contents to me.  When I do this in a Teams meeting, teammates are able to see files on the right side of the screen as expected.  I have tried both JAWS and Narrator with the same result.

### Expected Experience

When I select from the treeview on the left, I expect my screen reader to speak the contents that show on the right.

### Additional Context

none

**Comment by JasonYeMSFT** [comment]
@mblanchette5 Could you help us understand what is the desirable workflow that could help you understand what happens in the app? After selecting the resource in the tree view, are you expecting to hear the screen reader announce the files opened on the right hand side? Do you also expect Storage Explorer to transfer the control to the opened file so you can continue navigation there?

**Comment by mblanchette5** [comment]
Hi Jason, I was expecting both of those things.  But, since sighted people are able to see the list of files and I’m not hearing it, I assume that the issue is that the right side of the screen does not have focus, and I need to do something to set focus onto it.  I have tried all of the usual methods to do so, such as Tab, F6, F12, etc, along with some shortcuts that I  found in the article I  mentioned, but so far I haven’t had any luck.  If you can tell me what key sequence will put focus on the right side of the screen, I’d appreciate it.  Also, I was hoping to find a “Tips for Navigating Storage Explorer with a Screen Reader” article, like I’ve found before for Teams and others, but I haven’t found one.  If such a thing exists, I’d appreciate it if you could point me to it.  Thanks.

Matt Blanchette| Optum
Consulting Technologies
Optum Advisory Services Analytics

optum.com<https://hub.uhg.com/sites/hub/Optum/Resources/Brand/AppData/Local/Temp/www.optum.com>


This e-mail, including attachments, may include confidential and/or
proprietary information, and may be used only by the person or entity
to which it is addressed. If the reader of this e-mail is not the intended
recipient or intended recipient’s authorized agent, the reader is hereby
notified that any dissemination, distribution or copying of this e-mail is
p

**Comment by JasonYeMSFT** [comment]
@mblanchette5 I checked our old versions and this is definitely something we regressed for a long time. Sorry about that. I will fix this issue in our next release (1.32.0). I will share a private build here before we ship 1.32.0. If you can share some feedback on our fix it would be very appreciated.

**Comment by mblanchette5** [comment]
Hi Jason, do you know the newest  version that this issue did not exist in?  I ask because I may have an easier time if our company has already approved an earlier build that does not have this issue.  Thank you for the quick response.

Matt Blanchette| Optum
Consulting Technologies
Optum Advisory Services Analytics

optum.com<https://hub.uhg.com/sites/hub/Optum/Resources/Brand/AppData/Local/Temp/www.optum.com>


This e-mail, including attachments, may include confidential and/or
proprietary information, and may be used only by the person or entity
to which it is addressed. If the reader of this e-mail is not the intended
recipient or intended recipient’s authorized agent, the reader is hereby
notified that any dissemination, distribution or copying of this e-mail is
p

**Comment by JasonYeMSFT** [comment]
1.24.3 would be a good one to try. I just tested blob, file share, queue, table, adls gen2 explorers in it and I can see the focus being transferred from the node to the explorer when an explorer opens. I think prior versions should also work but they are more likely to have broken features due to the service updates.

**Comment by mblanchette5** [comment]
Great.  Thanks again.

Matt Blanchette| Optum
Consulting Technologies
Optum Advisory Services Analytics

optum.com<https://hub.uhg.com/sites/hub/Optum/Resources/Brand/AppData/Local/Temp/www.optum.com>


This e-mail, including attachments, may include confidential and/or
proprietary information, and may be used only by the person or entity
to which it is addressed. If the reader of this e-mail is not the intended
recipient or intended recipient’s authorized agent, the reader is hereby
notified that any dissemination, distribution or copying of this e-mail is
p

**Comment by JasonYeMSFT** [comment]
@mblanchette5 We made a 1.32.0 private build with a change to address your issue. We restored the behavior of "automatically transferring the focus from the selected tree view node to the opened explorer". Once the focus is transferred, the screen reader should be able to read the content in the opened explorer and you can navigate in the explorer right away.

[screen reader private build](https://storageexplorerpublish.blob.core.windows.net/privatebuilds/1.32.0-auto-focus-opened-explorer/StorageExplorer-windows-x64.exe)

**Comment by mblanchette5** [comment]
Hi JP, I’m forwarding this to you because I’m not sure what I should do here.

To sum up: the mentioned application allows you to look at data stored in Azure containers, and copy file(s) from one container and paste them to another one.  The issue for me is, I can get so far as selecting a container (for a sighted person, these are in a treeview on the left side of the screen), but after that, while a sighted person would then see a list of files and folders on the right side of the screen, I am not hearing JAWS tell me anything.  I’ve tried refreshing the screen like you taught me with Aha, but that isn’t helping.

I  opened a case with Microsoft, and found that this was an old issue  that was fixed, then re-broken in later builds, and they plan to fix it again in the next build.   However, in the meantime, they have sent me a link to a private build that they say will fix this.

The idea of a “private build” makes me nervous, and I don’t know where this fits in with ADA reasonable accommodations.  Can I get your thoughts on this?

Thanks.

**Comment by JasonYeMSFT** [comment]
@mblanchette5 Let me clarify what we use "private build" for. A private build is an instance of our application with a change that could potentially solve your problem. It's meant to be used to verify the fix and nothing beyond that. We've done the work to make sure that instance won't cause any problem while verifying the fix. In your case, to verify the fix, you can sign-in, navigate to a container node in the tree view, open it and see if the screen reader announces the opened explorer on the right-hand side and if you feel comfortable with the new experience. Once you tried that out, you can uninstall the private build and give us your feedback on the new experience.

**Comment by mblanchette5** [comment]
Hi Jason, I am sorry for not replying sooner.  I have downloaded the build, but I don’t have permission to install it.  I am working on getting this.

**Comment by JasonYeMSFT** [comment]
@mblanchette5 We are about to publish the release. If you haven't tried the private build that's fine. You don't need to worry about it once 1.32.0 releases. The 1.32.0 official release will contain exactly the same accessibility improvement we would like you to test in the private build and you can just upgrade to 1.32.0 to start using it.
